### ISSUE TITLE
DMCA Takedown Notice

### ISSUE THREAD

**Issue by EranSch** [Swingline0/openMultiLogin#20]
Well folks, it's been quite a ride but I'm getting tired of dealing with the Chrome Store on this. I received a notice that the listing for openMultiLogin (old URL above) was reported as copyright infringement, under the DMCA. Because this code is technically an infringement, it's not really something I feel like pushing back on, despite the fact that the reporter is obviously up to some shadiness.

Beware
---

Despite being the legitimate author, the reporter of the infringement hosts two semi-duplicate extension listings under the original name ("**MultiLogin**") and the name of this extension ("**openMultiLogin**"). The super shady part is that both listings actually use my description text, including links to this repo! (**NOTE**: If you arrived here from either of these listings, please be aware that I am not the author for either of the extensions below. **Use them at your own risk!**)

Shadiness Example # 1
----

![image](https://user-images.githubusercontent.com/1479206/38634758-7b8cc3b2-3d91-11e8-9615-c7667c5ded7b.png)

Shadiness Example # 2
----

![image](https://user-images.githubusercontent.com/1479206/38634780-87906d12-3d91-11e8-9bf0-88984716b2bb.png)

If I were you, **I would avoid both the listings above like the plague!**
-

Thank you, everyone who participated and strived to help push this strange open source experience along. I'll leave this repo up if you want to pull the source. Enjoy!

DMCA Takedown Request
---
>subject_lr_dmca: Уведомление направленное в Google 
full_name: Balas Viktor 
companyname: MultiLogin 
represented_copyright_holder: Balas Viktor 
contact_email_noprefill: multiloginplugin@gmail.com 
country_residence: UA 
description_of_copyrighted_work: Google chrome plugin source 
location_of_copyrighted_work: 
https://chrome.google.com/webstore/detail/multilogin/nknfhhmhoflkcijaodalbncnmidocced 
url_box: 
https://chrome.google.com/webstore/detail/openmultilogin/plaahcaagklllbcjognjgcnldgjnjhpb 
dmca_affirmations_authorized: agree 
dmca_affirmations_penalty: agree1 
signature: Balas Viktor 
hidden_product: chromewebstoreextensionsgallery 
hidden_ctx: 
geolocation: UA 
hidden_uraw:

**Comment by Damianonymous** [comment]
I do not understand it. How can the rightful owner refer to your repository where the source code of the owner is located?

**Comment by Damianonymous** [comment]
I do not understand it. How can the rightful owner send people back to your repository? Where is the owner's source code?

**Comment by alex-ries** [comment]
Based on the writing above what happened is: Swingline saw something useful (but seemingly sketchy) and rolled his own minus sketchiness. And the original author slammed him with DMCA take down notices. And it's not really worth fighting if this code kinda-sort-a-probably is a de-obfuscated version of the original work.

I'm still using this even though google insisting on putting up a notice that I have "Dev extensions loaded" makes it a minor pain in the neck. Totally worth not having some rando track my activity though.

**Comment by MasterOfUnlocking** [comment]
So is this it? The "rewrite" isn't coming anymore? What was supposed to be included in it? The extension seems to work as it is...

P.S. What does "de-obfuscation" mean, in layman terms?

**Comment by alex-ries** [comment]
@MasterOfUnlocking I can't speak to the first bit, but I can answer the de-obfuscation question. The easiest way to answer is to describe what code obfuscation is: it's using different types of patterns to make code unreadable, while leaving it functional. Think about things like: adding code lines that have no function, or using some extreme forms of short hand (renaming all functions/fields/methods/attributes to single letters, anonymous functions), it's basically a way to "hide" (obfuscate :) ) the original code base. So de-obfuscation is to put the code back in to a somewhat maintainable state trying to identify all the obfuscation lines and often also beautifying it so you can figure out what the code is doing. So what was done here (based on the de-obfuscation comment), the current code was created from a copy of the original (shady) extension which was then de-obfuscated. So technically this code is a copy.

I doubt doing a "re-write" based off this code will be seen reasonably as a completely new extension.

**Comment by Damianonymous** [comment]
Original extensions without open source code have also been removed from the Chrome Store. How can I remove a warning about loading an extension in development mode that appear when my browser is turned on?

**Comment by alex-ries** [comment]
@Damianonymous you can't, I'll see if I can find the thread (I may just not bother tbh, so maybe google it yourself :) ): but basically google's stance on this is that the warning being permanent and not something one can turn on is to protect us from ourselves. They're afraid of malware installing itself as a developer extension and (speculation) probably something about being able to easily create alternative stores/extensions that otherwise violate Google's Chrome Extension policies.

**Comment by caspertone2003** [comment]
@Damianonymous 
Yes, you can, at least by now.
https://stackoverflow.com/questions/30287907/how-to-get-rid-of-disable-developer-mode-extensions-pop-up/30361260#30361260

**Comment by Damianonymous** [comment]
@caspertone2003 
Thank you, it works. I needed it, I hope that the script will be updated :)
